courier/courier.go

```diff
@@ -16,27 +17,38 @@ import (
 
 	gomail "github.com/ory/mail/v3"
 
-	"github.com/ory/kratos/driver/config"
 	"github.com/ory/kratos/x"
 )
 
 type (
-	smtpDependencies interface {
+	SMTPConfig interface {
+		CourierSMTPURL() *url.URL
+		CourierSMTPFrom() string
+		CourierSMTPFromName() string
+		CourierSMTPHeaders() map[string]string
+		CourierTemplatesRoot() string
+	}
+	SMTPDependencies interface {
 		PersistenceProvider
 		x.LoggingProvider
-		config.Provider
+		CourierConfig(ctx context.Context) SMTPConfig
```

Comment:
Maybe make this it's own interface type?
```go
type CourierConfigProvider interface {
	CourierConfig(ctx context.Context) SMTPConfig
}
```
Might make it easier to handle elsewhere then.